src/v/cluster/tests/distributed_kv_stm_tests.cc

```diff
@@ -10,11 +10,43 @@
 #include "cluster/distributed_kv_stm.h"
 #include "raft/tests/raft_group_fixture.h"
 #include "raft/tests/simple_raft_fixture.h"
+#include "serde/envelope.h"
 
 static ss::logger logger{"kv_stm-test"};
 
-using test_key = ss::sstring;
-using test_value = kafka::offset;
+struct test_key
+  : serde::envelope<test_key, serde::version<0>, serde::compat_version<0>> {
+    test_key() = default;
+    // NOLINTNEXTLINE(hicpp-explicit-conversions)
+    test_key(int k)
+      : k(k) {}
+    int k;
+
+    template<typename H>
+    friend H AbslHashValue(H state, const test_key& k) {
+        return H::combine(std::move(state), k.k);
+    }
+    friend bool operator==(const test_key&, const test_key&) = default;
+    friend auto operator<=>(const test_key&, const test_key&) = default;
+    friend std::ostream& operator<<(std::ostream& os, const test_key& k) {
+        return os << k.k;
+    };
+
+    auto serde_fields() { return std::tie(k); }
+};
```

Comment:
do we need  struct wrapper with the switch to is_trivially_copyable.. feel free to ignore this comment, ok to leave as is.

Reply:
Just aliased both to `int`, I needed to kick CI anyways - thanks for this 😄